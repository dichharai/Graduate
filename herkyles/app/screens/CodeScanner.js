import React, { Component } from 'react';
import {
  Alert,
  Linking,
  Dimensions,
  LayoutAnimation,
  Text,
  View,
  StatusBar,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { BarCodeScanner, Permissions } from 'expo';

class CodeScanner extends Component {
  static navigationOptions = {
    title: "Code Scanner", 
    headerStyle: {
        backgroundColor: "#000000",
    }, 
      headerTintColor: '#facf33',
  };

  state = {
    hasCameraPermission: null,
    lastScannedUrl: null,
  };

  componentDidMount() {
    this.requestCameraPermission();
  }

  requestCameraPermission = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA);
    this.setState({
      hasCameraPermission: status === 'granted',
    });
  };

  handleBarCodeRead = result => {
    if (result.data !== this.state.lastScannedUrl && this.state.lastScannedUrl == null) {
      LayoutAnimation.spring();
      this.setState({ lastScannedUrl: result.data });
    }    
  };

  handlePressUrl = () => {
    if (!this.state.lastScannedUrl) {
      return;
    }

    Alert.alert(
      'Open this URL?',
      this.state.lastScannedUrl,
      [
        {
          text: 'Yes',
          onPress: () => {
            Linking.openURL(this.state.lastScannedUrl),
            this.setState({ lastScannedUrl: null })
          },
        },
        { 
          text: 'No',
          onPress: () => { 
            this.setState({ lastScannedUrl: null })
          },
        },
      ],
      { cancellable: false }
    );
  };

  render() {
    return (
      <View style={styles.container}>

        <Text style={{textAlign: 'center', fontSize: 15}}>{'Some machines have a QR code located on it. \n Scan the code using the scanner below to receive more information about the machine.\n\n\n\n'}</Text>
        
        {this.state.hasCameraPermission === null
          ? <Text>Requesting for camera permission</Text>
          : this.state.hasCameraPermission === false
              ? <Text style={{ color: '#fff' }}>
                  Camera permission is not granted
                </Text>
              : <View style={{height: 205, width:205, backgroundColor: 'black',alignItems: 'center', justifyContent: 'center',}}> 
                  <BarCodeScanner
                    onBarCodeRead={this.handleBarCodeRead}
                    style={{
                      height: 200,
                      width: 200,
                    }}
                  />
                </View>}

        {this.handlePressUrl()}

        <StatusBar hidden />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fff',
  },
});

export default CodeScanner
