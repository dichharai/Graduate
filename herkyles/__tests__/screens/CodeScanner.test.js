import React from "react";
import CodeScanner from "../../app/screens/CodeScanner";

import renderer from "react-test-renderer";

describe ("Code Scanner", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<CodeScanner />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});