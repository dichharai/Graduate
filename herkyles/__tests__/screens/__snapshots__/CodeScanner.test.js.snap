// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Scanner renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Some machines have a QR code located on it. 
 Scan the code using the scanner below to receive more information about the machine.




  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Requesting for camera permission
  </Text>
</View>
`;
